### PR TITLE
Finer-grain control of exception behaviour in view renderers

### PR DESCRIPTION
**Pull request by acwwat**
This PR is one possible implementation of issue #1811. In short, I am standardizing the exception thrown by each `ViewRenderer` implementation as `ViewRenderException`, which is then caught and rethrown as `WebApplicationException` by `ViewMessageBodyWriter`.

To preserve previous behavior of returning an HTML error response on render exceptions, I have added a new `ViewRenderExceptionMapper` class which should be registered. Consequently I am still changing the original behavior by adding this registration requirement. Let me know if this is OK from backward-compatibility standpoint - if needed we can register the mapper in `ViewBundle`, although I don't know how I can register another exception mapper afterwards if I want custom behaviour. This also requires change to the documentation.

Otherwise, one can choose to provide their own `ExceptionMapper` to catch `WebApplicationException` and check that the cause is `ViewRenderException`, then do whatever is needed with its cause in the form of an exception thrown by the rendering code (Viewmarker, Mustache, etc.)

I tried updating the tests to verify both cases of with and without `ViewRenderExceptionMapper`, but I can't seem to get registration in each test case working like this:

` target("/test/bad").regsiter(new ViewRenderExceptionMapper()).request().get(String.class);`

I had to register the mapper in `configure()`. Please let me know if you have any solution to this so I can increase the test coverage.

Thanks.

### CODE REVIEW

**Comment by nickbabcock** [comment]
I think it would be worthwhile to have a test in `dropwizard-e2e/src/main/java/com/example/app1` showing that the behavior can be changed.

**Comment by acwwat** [comment]
@nickbabcock I added a test/example to e2e on overriding the standard 503 response on missing mustache view with a 404 response.

File: dropwizard-views/src/main/java/io/dropwizard/views/ViewMessageBodyWriter.java [hunk 1]
```diff
@@ -82,12 +76,9 @@ public void writeTo(View t,
                 }
             }
             throw new ViewRenderException("Unable to find a renderer for " + t.getTemplateName());
-        } catch (Exception e) {
+        } catch (ViewRenderException e) {
             LOGGER.error("Template Error", e);
```
Comment (dren-dk) [inline dropwizard-views/src/main/java/io/dropwizard/views/ViewMessageBodyWriter.java]: My preference would be to format the log message in my own ExceptionMapper, so could this go into the ExceptionMapper as well?

Comment (acwwat) [inline dropwizard-views/src/main/java/io/dropwizard/views/ViewMessageBodyWriter.java]: That makes sense. I've removed the logging from `ViewMessageBodyWriter` and added the logging code to `ViewRenderExceptionMapper` to preserve the original exception logging behavior.

**Comment by arteam** [comment]
The pull request looks good to me. I don't think the code coverage is so important, what matters you provided relevant tests. And thank you very much for a very detailed documentation for `ViewRenderException`!

**Pull request merged**